### ISSUE TITLE
Can't JIT-compile `asm_propagate` with option `output_dx`

### ISSUE THREAD

**Issue by tomsturges** [chromatix-team/chromatix#176]
I have some simple code below to train a sequence of phase masks.
It uses `asm_propagate`. It works.
However, when I use the option `output_dx` the `jit` fails.
I would like to have a different `output_dx` because I want the loss function to see a smaller region around the optical axis with higher resolution. It seems to me that this could in principle be made `jit` compatible, but I can't get it to work.
Suggestions please?

Thanks.

```python
import chromatix as cx
from chromatix.functional import gaussian_plane_wave, asm_propagate, Field, phase_change
import matplotlib.pyplot as plt
import numpy as np
import cmcrameri.cm as cmc
from bst import generate_bst2D
import equinox as eqx
from jaxtyping import Array, Float, Int
import jax
import jax.numpy as jnp
import my_little_helpers as mlh
import optax

def STATIC(): return eqx.field(static=True)
def PARAM():  return eqx.field(static=False)

class Model(eqx.Module):
    # Trainables parameters
    phases: list[Array] = PARAM()
    # Static
    n_layers: int = STATIC()
    shape: tuple[int, int] = STATIC()
    dx: float = STATIC()
    spectrum: float | np.ndarray = STATIC()
    z: float = STATIC()
    n: float = STATIC()
    pad_width: int = STATIC()
    waist: float = STATIC()
    kykx: float = STATIC()

    def __call__(self) -> Field:
        field = gaussian_plane_wave(shape=self.shape, dx=self.dx, spectrum=self.spectrum, waist=self.waist)
        for phase in self.phases:
            field = phase_change(field, phase)
            field = asm_propagate(field, z=self.z, n=self.n, pad_width=self.pad_width)

        return field

SHAPE = (500, 500)
NMASKS = 3

phases = [jnp.zeros(SHAPE) for _ in range(NMASKS)]
model = Model(phases=phases, n_layers=NMASKS, shape=SHAPE, dx=0.1, spectrum=0.45, n=1.0, z=100, waist=14, kykx=(0, 0), pad_width=0)

field = model()
grid = field.grid
X = grid[..., 0]
Y = grid[..., 1]
R = jnp.sqrt(X**2 + Y**2)
mask = (R < 2).astype(jnp.float32)

params, static = eqx.partition(model, True)

def loss(params, static):
    model = eqx.combine(params, static)
    field = model() # type: Field
    I = field.intensity
    return -jnp.sum(I * mask)

optim = optax.adam(1e-3)
opt_state = optim.init(eqx.filter(model, True))

@eqx.filter_jit
def update(model, opt_state):
    loss_value, grads = eqx.filter_value_and_grad(loss)(params, static)
    updates, opt_state = optim.update(grads, opt_state, eqx.filter(model, True))
    model = eqx.apply_updates(model, updates)
    return model, opt_state, loss_value

fval = []
for _ in range(100):
    model, opt_state, loss_value = update(model, opt_state)
    fval.append(loss_value)

I = model().intensity
plt.imshow(I)

```

**Comment by diptodip** [comment]
Fixed by PR #178.

**Comment by tomsturges** [comment]
Thanks very much, this works for me now.
